packages/frames.js/src/lens/index.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  FrameVerifySignatureResult,
+  LensClient,
+  development,
+} from "@lens-protocol/client";
+
+type LensFrameMessageType = {
+  clientProtocol: string;
+  untrustedData: {
+    specVersion: string;
+    profileId: string;
+    pubId: string;
+    url: string;
+    buttonIndex: number;
+    unixTimestamp: number;
+    deadline: number;
+    inputText: string;
+    state: string;
+    actionResponse: string;
+  };
+  trustedData: {
+    messageBytes: string;
+    identityToken: string;
+  };
+};
+
+export type LensFrameMessageReturnType = LensFrameMessageType & {
+  isActionVerifed: boolean;
+};
+
+export function isLensFrameActionPayload(
+  frameActionPayload: unknown
+): frameActionPayload is LensFrameMessageType {
+  return (
+    typeof frameActionPayload === "object" &&
+    frameActionPayload !== null &&
+    "clientProtocol" in frameActionPayload &&
+    typeof frameActionPayload.clientProtocol === "string" &&
+    frameActionPayload.clientProtocol.startsWith("lens@")
+  );
+}
+
+export async function getLensFrameMessage(
+  frameActionPayload: LensFrameMessageType
+): Promise<LensFrameMessageReturnType> {
+  const lensClient = new LensClient({
+    environment: development,
+  });
+
+  const typedData = await lensClient.frames.createFrameTypedData({
+    ...frameActionPayload.untrustedData,
+  });
+
+  const response = await lensClient.frames.verifyFrameSignature({
+    identityToken: frameActionPayload.trustedData.identityToken,
+    signature: frameActionPayload.trustedData.messageBytes,
+    signedTypedData: typedData,
+  });
+
+  return {
+    ...frameActionPayload,
+    isActionVerifed: response === FrameVerifySignatureResult.Verified,
+  };
```

Comment:
```suggestion
  return {
    ...frameActionPayload,
    isValid: response === FrameVerifySignatureResult.Verified,
  };
```

- this should return decoded trusted data
- currently the XMTP open frames library throws an error if the message isn't valid - we need to decide if we want the standard in frames.js to be throwing an error or returning `isValid = false`, like the farcaster verification does currently. do you have thoughts here? cc @michalkvasnicak @davidfurlong. if we decide to go with a return value then we should stick to `isValid`

Reply:
I modified `getLensFrameMessage` to return `typedData.value` which is the payload that is verified based on the signature (`messageBytes`), and changed the flag name to `isValid`.